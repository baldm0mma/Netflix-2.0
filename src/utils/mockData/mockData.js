export const mockMovies = {
  popular: {
    "id": 420818,
    "video": false,
    "vote_average": 7.1,
    "title": "The Lion King",
    "popularity": 636.278,
    "poster_path": "\/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg",
    "original_language": "en",
    "original_title": "The Lion King",
    "genre_ids": [
      12,
      16,
      10751,
      18,
      28
    ],
    "backdrop_path": "\/1TUg5pO1VZ4B0Q1amk3OlXvlpXV.jpg",
    "adult": false,
    "overview": "Simba idolises his father, King Mufasa, and takes to heart his own royal destiny. But not everyone in the kingdom celebrates the new cub's arrival. Scar, Mufasa's brother—and former heir to the throne—has plans of his own. The battle for Pride Rock is ravaged with betrayal, tragedy and drama, ultimately resulting in Simba's exile. With help from a curious pair of newfound friends, Simba will have to figure out how to grow up and take back what is rightfully his.",
    "release_date": "2019-07-12"
  }
};

export const mockMovie = {
  "id": 420818,
  "video": false,
  "vote_average": 7.1,
  "title": "The Lion King",
  "popularity": 636.278,
  "poster_path": "\/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg",
  "original_language": "en",
  "original_title": "The Lion King",
  "genre_ids": [
    12,
    16,
    10751,
    18,
    28
  ],
  "backdrop_path": "\/1TUg5pO1VZ4B0Q1amk3OlXvlpXV.jpg",
  "adult": false,
  "overview": "Simba idolises his father, King Mufasa, and takes to heart his own royal destiny. But not everyone in the kingdom celebrates the new cub's arrival. Scar, Mufasa's brother—and former heir to the throne—has plans of his own. The battle for Pride Rock is ravaged with betrayal, tragedy and drama, ultimately resulting in Simba's exile. With help from a curious pair of newfound friends, Simba will have to figure out how to grow up and take back what is rightfully his.",
  "release_date": "2019-07-12"
}

export const mockUser = {
  email: 'happy@sad.com',
  password: 'ilikeburritos',
  id: 2,
  name: 'coolGui420'
}

export const mockGenre = 'popular';

export const mockPost = {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({
    user: mockUser
  })
}

export const mockLoading = false;

export const mockFavorited = true;

export const mockFavorites = [{
  "id": 420818,
  "video": false,
  "vote_average": 7.1,
  "title": "The Lion King",
  "popularity": 636.278,
  "poster_path": "\/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg",
  "original_language": "en",
  "original_title": "The Lion King",
  "genre_ids": [
    12,
    16,
    10751,
    18,
    28
  ],
  "backdrop_path": "\/1TUg5pO1VZ4B0Q1amk3OlXvlpXV.jpg",
  "adult": false,
  "overview": "Simba idolises his father, King Mufasa, and takes to heart his own royal destiny. But not everyone in the kingdom celebrates the new cub's arrival. Scar, Mufasa's brother—and former heir to the throne—has plans of his own. The battle for Pride Rock is ravaged with betrayal, tragedy and drama, ultimately resulting in Simba's exile. With help from a curious pair of newfound friends, Simba will have to figure out how to grow up and take back what is rightfully his.",
  "release_date": "2019-07-12"
}]

export const mockError = 'Something went wrong';

export const mockGenres = [{
    "id": 28,
    "name": "Action",

  },
  {
    "id": 12,
    "name": "Adventure",

  }, {
    "id": 16,
    "name": "Animation",

  }, {
    "id": 35,
    "name": "Comedy",

  }, {
    "id": 80,
    "name": "Crime",

  }, {
    "id": 99,
    "name": "Documentary",

  }, {
    "id": 18,
    "name": "Drama",

  }, {
    "id": 10751,
    "name": "Family",

  }, {
    "id": 14,
    "name": "Fantasy",

  }, {
    "id": 36,
    "name": "History",

  }, {
    "id": 27,
    "name": "Horror",

  }, {
    "id": 10402,
    "name": "Music",

  }, {
    "id": 9648,
    "name": "Mystery",

  }, {
    "id": 10749,
    "name": "Romance",

  }, {
    "id": 878,
    "name": "Science Fiction",

  }, {
    "id": 10770,
    "name": "TV Movie",

  }, {
    "id": 53,
    "name": "Thriller",

  }, {
    "id": 10752,
    "name": "War",

  }, {
    "id": 37,
    "name": "Western",

  }
]