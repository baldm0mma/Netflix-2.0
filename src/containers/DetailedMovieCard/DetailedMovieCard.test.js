import {
	DetailedMovieCard,
	mapStateToProps,
	mapDispatchToProps
} from './DetailedMovieCard';
import {
	shallow
} from 'enzyme';
import React from 'react';
import {
	fetchSingleMovie
} from '../../utils/API/ApiFetch';
import {mockMovie} from '../../utils/mockData/mockData';

jest.mock('../../utils/API/ApiFetch');

describe('DetailedMovieCard', () => {
		let wrapper, instance;

		beforeEach(() => {
			wrapper = shallow( < DetailedMovieCard specificMovie={mockMovie}/> );
			instance = wrapper.instance();
		});
		it('should match snapshot', () => {
	describe('component', () => {
			expect(wrapper).toMatchSnapshot();
		});
	});
	describe('CDM', () => {
		it('should invoke fetchSingleMovie', () => {
			instance.componentDidMount();
			expect(fetchSingleMovie).toHaveBeenCalledWith(mockMovie.id);
		});

		it('should set state to currentMovie', async () => {
			await instance.componentDidMount();
			expect(wrapper.state('currentMovie')).toEqual();
		});
		describe('mapStateToProps', () => {
			it('should return an object with the user and specificMovie', () => {
				const mockState = {
					specificMovie: {
						title: 'mockTitle',
						id: 1
					}
				};
				const mappedProps = mapStateToProps(mockState);
				expect(mappedProps).toEqual(mockState);
			});
		});
		describe('mapDispatchToProps', () => {
			it('should call dispatch with a fetchSingleMovie action when fetchSingleMovie is called', () => {
				const mockDispatch = jest.fn();
				const actionToDispatch = fetchSingleMovie(1);
				const mappedProps = mapDispatchToProps(mockDispatch);
				mappedProps.fetchSingleMovie(1);
				expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
			});
		});
	});
});