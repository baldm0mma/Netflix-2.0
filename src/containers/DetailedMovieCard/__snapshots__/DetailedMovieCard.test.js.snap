// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedMovieCard component should match snapshot 1`] = `ShallowWrapper {}`;
